src/platform/forms-system/test/js/fields/ObjectField.unit.spec.jsx

```diff
@@ -443,4 +443,98 @@ describe('Schemaform: ObjectField', () => {
     expect(id0).to.equal('root_test__title');
     expect(id1).to.equal('root_test2__title');
   });
+  it('should render with a fieldset when forceDivWrapper is false', () => {
+    const onChange = sinon.spy();
+    const onBlur = sinon.spy();
+    const schema = {
+      type: 'object',
+      properties: {
+        prop1: {
+          type: 'boolean',
+        },
+        prop2: {
+          type: 'boolean',
+        },
+        prop3: {
+          type: 'boolean',
+        },
+        prop4: {
+          type: 'boolean',
+        },
+      },
+    };
+    const uiSchema = {
+      'ui:title': 'Test title',
+      'ui:options': {
+        forceDivWrapper: false,
+      },
+      prop1: { 'ui:title': 'title1' },
+      prop2: { 'ui:title': 'title2' },
+      prop3: { 'ui:title': 'title3' },
+      prop4: { 'ui:title': 'title4' },
+    };
+
+    const form = ReactTestUtils.renderIntoDocument(
+      <div>
+        <ObjectField
+          uiSchema={uiSchema}
+          schema={schema}
+          idSchema={{}}
+          formData={{}}
+          onChange={onChange}
+          onBlur={onBlur}
+        />
+      </div>,
+    );
+    const formDOM = getFormDOM(form);
+    const fieldsets = formDOM.querySelectorAll('fieldset');
```

Comment:
Should this check `'fieldset + legend'`? that way we know for sure the `legend` is in place, or maybe that's out of scope for this change?

It might also be a good idea to make one of the `ui:titles` a function instead of a string.